include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -411,22 +411,9 @@ SECTIONS
 
 #if defined(CONFIG_APP_SHARED_MEM)
 #define APP_SHARED_ALIGN . = ALIGN(_region_min_align);
-#if defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
-		#include <app_smem.ld>
-#else
-	SECTION_PROLOGUE(_APP_SMEM_SECTION_NAME, (OPTIONAL),)
-	{
-		APP_SHARED_ALIGN;
-		_app_smem_start = .;
-		APP_SMEM_SECTION()
-		/* Align the end of shared app mem section with the
-		 * minimum granularity required by MPU.
-		 */
-		APP_SHARED_ALIGN;
-		_app_smem_end = .;
+#define SMEM_PARTITION_ALIGN MPU_ALIGN
```

Comment:
Unless I'm missing something you are now enforcing power of two alignment on all ARM targets

Reply:
No not really . The alignement is done by a macro out side app_smem.ld. If we have a power of 2 MPU the alignment computed is different compared to a NXP MPU.